File: app/src/main/java/com/psliusar/layers/sample/screen/listener/ListenerViewModel.java
package com.psliusar.layers.sample.screen.listener;

import android.app.Activity;
import android.graphics.Bitmap;

import com.psliusar.layers.LayersActivity;
import com.psliusar.layers.ViewModel;

import androidx.annotation.NonNull;

public class ListenerViewModel extends ViewModel {

    private final ListenerModel model = new ListenerModel();

    void getPhoto(@NonNull LayersActivity activity, @NonNull ListenerModel.Updatable<Bitmap> updatable) {
        manage(model.getPhotoUri(activity, updatable));
    }

    void takePhotoClick(@NonNull Activity activity) {
        model.requestPhoto(activity);
    }
}

File: app/src/main/java/com/psliusar/layers/sample/screen/listener/ListenerViewModel.kt
package com.psliusar.layers.sample.screen.listener

import android.app.Activity
import android.graphics.Bitmap
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import com.psliusar.layers.LayersActivity

class ListenerViewModel : ViewModel() {

    private val _photoUri = MutableLiveData<Bitmap>()
    val photoUri: LiveData<Bitmap> = _photoUri

    private val model = ListenerModel()

    fun getPhoto(activity: LayersActivity, updatable: ListenerModel.Updatable<Bitmap>) {
        //manage(model.getPhotoUri(activity, updatable))


    }

    fun takePhotoClick(activity: Activity) {
        model.requestPhoto(activity)
    }
}
